src/core/Directus/Database/Schema/SchemaFactory.php

```diff
@@ -160,14 +160,14 @@ public function alterTable($name, array $data)
         }
 
         $toChangeColumnsData = ArrayUtils::get($data, 'change', []);
-
+        
         // Throws an exception when trying to make the field required and there are items with no value for that field in collection
         foreach ($toChangeColumnsData as $column) {
             if($column['required']) {
-                $selectQuery = sprintf('SELECT * FROM `%s` WHERE `%s` IS NULL',$name,$column['field']);
+                $selectQuery = sprintf('SELECT COUNT(*) as "count" FROM `%s` WHERE `%s` IS NULL',$name,$column['field']);
```

Comment:
Is there a reason why we're not using query builder here? I'm not sure where those parameters come from, and I'm not confident on merging this because of security issues.

Is it possible to change it to use a query builder instead?